Geometry/DTGeometryBuilder/plugins/DTGeometryValidate.cc

```diff
@@ -82,11 +82,12 @@ class DTGeometryValidate : public one::EDAnalyzer<> {
 
 DTGeometryValidate::DTGeometryValidate(const edm::ParameterSet& iConfig)
     : dtGeometryToken_{esConsumes<DTGeometry, MuonGeometryRecord>(edm::ESInputTag{})},
-      infileName_(iConfig.getUntrackedParameter<string>("infileName", "cmsGeom10.root")),
+      infileName_(
+          iConfig.getUntrackedParameter<string>("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root")),
```

Comment:
You need to use FileInPath to get the file e.g. for python module you need to do something like
https://github.com/cms-sw/cmssw/blob/02d4198c0b6615287fd88e9a8ff650aea994412e/CalibCalorimetry/HcalPlugins/python/Hcal_FakeConditions_cff.py#L13

and from cc code you can do something like
https://github.com/cms-sw/cmssw/blob/0f58e3682e346587094bb503ef4556a4ed472d31/DetectorDescription/Core/test/DDIsValid.cppunit.cc#L172
OR
https://github.com/cms-sw/cmssw/blob/ed1b00288c95ceccee10546314d2384ee98ce063/RecoLocalMuon/GEMRecHit/plugins/GEMRecHitProducer.cc#L37

Reply:
@smuzaffar : I modified the constructor of the DTGeometryValidate.cc in this way:

..
 edm::FileInPath fp("Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root");                                                                                                                                   fwGeometry_.loadMap(fp.fullPath().c_str());

instead of 
...
 iConfig.getUntrackedParameter<string>("infileName", "Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root"))
....
fwGeometry_.loadMap(infileName_.c_str()); 

The compilation is ok, but when I run: "cmsRun Geometry/DTGeometryBuilder/test/pyton/validateDTGeometry_cfg.py" I see:

----- Begin Fatal Exception 27-Aug-2020 15:33:18 CEST-----------------------
An exception of category 'FileInPathError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=DTGeometryValidate label='valid'
Exception Message:
edm::FileInPath unable to find file Geometry/DTGeometryBuilder/data/cmsRecoGeom-2021.root anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /build/slomeo/CMSSW_11_2_X_2020-08-23-2300/poison:/build/slomeo/CMSSW_11_2_X_2020-08-23-2300/src:/build/slomeo/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/poison:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/src:/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-08-23-2300/external/slc7_amd64_gcc820/data
Current directory is: /build/slomeo/CMSSW_11_2_X_2020-08-23-2300/src
----- End Fatal Exception -------------------------------------------------

Where is my mistake?